### PR TITLE
Add Rails setup to Readme

### PR DESCRIPTION
**Pull request by testbrian**
Changed `ENV['GITHUB_KEY'], ENV['GITHUB_SECRET']` to `ENV['GITHUB_CLIENT_ID'], ENV['GITHUB_CLIENT_SECRET']` to match Github credentials and added a little info.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,20 +6,36 @@ on the [GitHub Applications Page](https://github.com/settings/applications).
 
 ## Basic Usage
 
-    use OmniAuth::Builder do
-      provider :github, ENV['GITHUB_KEY'], ENV['GITHUB_SECRET']
-    end
+Add to your `Gemfile`:
+gem 'omniauth-github'
```
Comment (jasonnoble) [inline README.md]: Add a blank line above and 4 space indent this to make MD format it as code?

    gem 'omniauth-github'

File: README.md [hunk 2]
```diff
@@ -6,20 +6,36 @@ on the [GitHub Applications Page](https://github.com/settings/applications).
...
 
-## Github Enterprise Usage
+Then `bundle install`
+
+Add to an initializer `config/initializers/omniauth.rb`:
```
Comment (jasonnoble) [inline README.md]: Create an initializer...

File: README.md [hunk 3]
```diff
@@ -6,20 +6,36 @@ on the [GitHub Applications Page](https://github.com/settings/applications).
...
+```
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :github, ENV['GITHUB_CLIENT_ID'], ENV['GITHUB_CLIENT_SECRET']
+end
+```
+## Github Callback
+
+When a user visits `/auth/github` and are redirected and signin at Github they are redirected to the callback specified in the Github application settings.
```
Comment (jasonnoble) [inline README.md]: When a user on your site visits `/auth/github` they will be redirected to Github and asked to verify permissions. When they accept, they will be sent back to your site based on the callback setting for the application you setup.

Perhaps a link to some Github documentation on how to add a Oauth application?  

Also may want to put something in here about using 127.0.0.1:3000 vs. localhost:3000 when doing local work with Github Oauth?  (Hint: 127.0.0.1 works, localhost will not.)

File: README.md [hunk 4]
```diff
@@ -6,20 +6,36 @@ on the [GitHub Applications Page](https://github.com/settings/applications).
...
 
-    provider :github, ENV['GITHUB_KEY'], ENV['GITHUB_SECRET'],
-        {
-          :client_options => {
-            :site => 'https://github.YOURDOMAIN.com/',
-            :authorize_url => 'https://github.YOURDOMAIN.com/login/oauth/authorize',
-            :token_url => 'https://github.YOURDOMAIN.com/login/oauth/access_token',
-          }
-        }
+The response to Rails in `env.omniauth`:
```
Comment (jasonnoble) [inline README.md]: When the user is redirected back to your site, the callback URL will have the following variables available...?

**Comment by jasonnoble** [comment]
Maybe we should add deprecation notices for `ENV['GITHUB_KEY'], ENV['GITHUB_SECRET']`?

**Comment by testbrian** [comment]
There isn't a deprecation AFAIK, just a slight rename so a deprecation warning wouldn't be necessary.

**Comment by matkoniecz** [comment]
@testbrian Thanks, this guide is very helpful. It is unfortunate that this PR was not accepted as it is currently a bit hard to find.

**Comment by testbrian** [comment]
How about merging then?

**Comment by tmilewski** [comment]
@matkoniecz @testbrian Hey guys! I just got added to the repo and I'm going through old PRs.
Thank you for this!

Considering that Omniauth and this gem, in particular, isn't specific to Rails, I'd be hesitant to add this entire setup to the README. With this said it'd work perfectly as a Wiki article which we can link to in the README.

I've set up a page here: https://github.com/intridea/omniauth-github/wiki/Rails-Usage

What do you think?

**Comment by tmilewski** [comment]
Moving to close.

**Pull request closed without merging**